invenio_app_rdm/records_ui/views/records.py

```diff
@@ -9,26 +9,61 @@
 
 """Routes for record-related pages provided by Invenio-App-RDM."""
 
-from functools import wraps
+from os.path import splitext
 
-from flask import abort, current_app, g, render_template, request
+from flask import abort, current_app, g, render_template, url_for
 from invenio_base.utils import obj_or_import_string
-from invenio_files_rest.views import ObjectResource
+from invenio_previewer.extensions import default
+from invenio_previewer.proxies import current_previewer
+from invenio_rdm_records.proxies import current_rdm_records
 from invenio_rdm_records.resources.serializers import UIJSONSerializer
 
-from ..utils import previewer_record_file_factory
-from .decorators import pass_record
+from .decorators import pass_file_metadata, pass_record_files, pass_record
+
+
+class PreviewFile:
```

Comment:
this one replaces `invenio_previewer.api:PreviewFile`, which sometimes has problems when accessing `.size` (because often, `self.file` would not support dict-like access)

this should probably find a better home

Reply:
Ok for now...... we may need to rethink all the file previewing a bit.....not very pretty to download a 20MB PDF in the browser :-)